lxd/storage/backend_lxd.go

```diff
@@ -0,0 +1,353 @@
+}
+
+func (b *lxdBackend) CreateInstanceFromImage(i Instance, fingerprint string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceFromMigration(i Instance, conn *websocket.Conn, args migration.SinkArgs, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstance(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstance(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MigrateInstance(i Instance, snapshots bool, args migration.SourceArgs) (migration.StorageSourceDriver, error) {
+	return nil, ErrNotImplemented
+}
+
+func (b *lxdBackend) RefreshInstance(i Instance, src Instance, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) BackupInstance(i Instance, targetPath string, optimized bool, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceUsage(i Instance) (uint64, error) {
+	return 0, ErrNotImplemented
+}
+
+func (b *lxdBackend) SetInstanceQuota(i Instance, quota uint64) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceDisk(i Instance) (string, string, error) {
+	return "", "", ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceSnapshot(i Instance, name string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstanceSnapshot(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RestoreInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateCustomVolume(vol api.StorageVolumesPost, op *operations.Operation) error {
+	project := "default"
+
+	volType, err := VolumeTypeNameToType(vol.Type)
+	if err != nil {
+		return err
+	}
+
+	// Create database entry for new storage volume.
+	err = VolumeDBCreate(b.state, b.name, vol.Name, vol.Description, vol.Type, false, vol.Config)
+	if err != nil {
+		return err
+	}
+
+	revert := true
```

Comment:
Should be called `revertDatabase`. I've recently gone with an approach of having one of those revert booleans per reversible action so it's easier to understand what's going on and avoids some of the issues we've had with sub-reverts in zfs.

Reply:
@stgraber ok will do. Out of interest, is there a reason we dont use transactions on the database and wait until the on disk stuff has happened before committing the transaction for the whole operation. Does this cause locking issues with sqlite's table level locking, or is it that it would require passing around the same db handle so different functions can read their own writes inside the transaction (to get IDs etc).

Reply:
OK will leave it out.